src/test_utils.rs

```diff
@@ -235,6 +239,54 @@ impl TestContext {
             .unwrap();
         Chat::load_from_db(self, chat_id).await.unwrap()
     }
+
+    /// Sends out the text message. If the other side shall receive it, you have to call `recv_msg()` with the returned `SentMessage`.
+    pub async fn send_text(&self, chat_id: ChatId, txt: &str) -> SentMessage {
+        let mut msg = Message::new(Viewtype::Text);
+        msg.set_text(Some(txt.to_string()));
+        chat::prepare_msg(&self, chat_id, &mut msg).await.unwrap();
+        chat::send_msg(&self, chat_id, &mut msg).await.unwrap();
+        self.pop_sent_msg().await
+    }
+
+    /// You can use this to debug your test by printing a chat structure
+    #[allow(dead_code)]
+    pub async fn print_chat(&self, chat: &Chat) {
+        let msglist = chat::get_chat_msgs(&self, chat.get_id(), 0x1, None).await;
```

Comment:
we don't have a constant for the `0x1`?

Reply:
I didn't think about this code a lot, I just copied it from `cmdline/repl.rs` and removed everything that gave compile errors :) (I mean, it's not even used regularly, only thought for a quick glimpse at a test problem).

But if we do say that we want nice code quality here, we may want to also think about removing that code duplication. We could probably both compile it for the repl via a feature and for tests via `cfg(test)`, or we make it a feature of the `deltachat` crate that you can get a string representation of a chat and just always compile it.

Reply:
> But if we do say that we want nice code quality here, we may want to also think about removing that code duplication. We could probably both compile it for the repl via a feature and for tests via `cfg(test)`, or we make it a feature of the `deltachat` crate that you can get a string representation of a chat and just always compile it.

You could make it a public function on the crate, it won't be compiled into deltachat-ffi as long as you don't expose it there so is fairly cheap from that point of view.

Reply:
though that's probably too much for this this PR I'd guess.

Reply:
ok, so I put a comment there so that we can maybe do this in a different PR.